framework/framework.go

```diff
@@ -206,144 +204,13 @@ func ProcessDelete(ctx context.Context, obj interface{}, resources []Resource) e
-				return nil
-			}
-			if resourcecanceledcontext.IsCanceled(ctx) {
-				ctx = resourcecanceledcontext.NewContext(ctx, make(chan struct{}))
-				continue
-			}
-
-			meta, ok := loggermeta.FromContext(ctx)
-			if ok {
-				meta.KeyVals["function"] = "GetDesiredState"
-				defer delete(meta.KeyVals, "function")
-			}
-			desiredState, err = r.GetDesiredState(ctx, obj)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-		}
-
-		var patch *Patch
-		{
-			if reconciliationcanceledcontext.IsCanceled(ctx) {
-				return nil
-			}
-			if resourcecanceledcontext.IsCanceled(ctx) {
-				ctx = resourcecanceledcontext.NewContext(ctx, make(chan struct{}))
-				continue
-			}
-
-			meta, ok := loggermeta.FromContext(ctx)
-			if ok {
-				meta.KeyVals["function"] = "NewDeletePatch"
-				defer delete(meta.KeyVals, "function")
-			}
-			patch, err = r.NewDeletePatch(ctx, obj, currentState, desiredState)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-		}
-
-		{
-			if reconciliationcanceledcontext.IsCanceled(ctx) {
-				return nil
-			}
-			if resourcecanceledcontext.IsCanceled(ctx) {
-				ctx = resourcecanceledcontext.NewContext(ctx, make(chan struct{}))
-				continue
-			}
-
-			if patch != nil {
-				createChange, ok := patch.getCreateChange()
-				if ok {
-					meta, ok := loggermeta.FromContext(ctx)
-					if ok {
-						meta.KeyVals["function"] = "ApplyCreateChange"
-						defer delete(meta.KeyVals, "function")
-					}
-					err := r.ApplyCreateChange(ctx, obj, createChange)
-					if err != nil {
-						return microerror.Mask(err)
-					}
-				}
-			}
-		}
-
-		{
-			if reconciliationcanceledcontext.IsCanceled(ctx) {
-				return nil
-			}
-			if resourcecanceledcontext.IsCanceled(ctx) {
-				ctx = resourcecanceledcontext.NewContext(ctx, make(chan struct{}))
-				continue
-			}
-
-			if patch != nil {
-				deleteChange, ok := patch.getDeleteChange()
-				if ok {
-					meta, ok := loggermeta.FromContext(ctx)
-					if ok {
-						meta.KeyVals["function"] = "ApplyDeleteChange"
-						defer delete(meta.KeyVals, "function")
-					}
-					err := r.ApplyDeleteChange(ctx, obj, deleteChange)
-					if err != nil {
-						return microerror.Mask(err)
-					}
-				}
-			}
+		err := r.EnsureDeleted(ctx, obj)
+		if err != nil {
+			return microerror.Mask(err)
 		}
 
-		{
-			if reconciliationcanceledcontext.IsCanceled(ctx) {
-				return nil
-			}
-			if resourcecanceledcontext.IsCanceled(ctx) {
-				ctx = resourcecanceledcontext.NewContext(ctx, make(chan struct{}))
-				continue
-			}
-
-			if patch != nil {
-				updateChange, ok := patch.getUpdateChange()
-				if ok {
-					meta, ok := loggermeta.FromContext(ctx)
-					if ok {
-						meta.KeyVals["function"] = "ApplyUpdateChange"
-						defer delete(meta.KeyVals, "function")
-					}
-					err := r.ApplyUpdateChange(ctx, obj, updateChange)
-					if err != nil {
-						return microerror.Mask(err)
-					}
-				}
-			}
+		if reconciliationcanceledcontext.IsCanceled(ctx) {
+			return nil
 		}
```

Comment:
Note handling reconciliationcanceledcontext is here, but handling resourcecanceledcontext is in the crud_resource.go

For simplified Resource resourcecanceledcontext doesn't make sense.

Reply:
Yeah, when you want to cancel the simple resource you return and be done.

Reply:
Update `reconciliationcanceledcontext` is handled in `crud_resource.go` as well, because it cancels between operations. Like between `GetCurrentState` and `GetDesiredState`.